### DISCUSSION TITLE
Compressing uncompressed repo/archives

### DISCUSSION THREAD

**Discussion by DizkoDan** [borgbackup/borg#6714] [Q&A] [answered]
Hi,

I just started using borg (and borgmatic) in a few places, and so far it's awesome! However, on my larger setup, which is from a NAS to a remote backup server, I initially set up the repo uncompressed. I'm now wondering if it's possible/advisable to compress them, and change the config to use compression going forward? I saw the `recreate` command & the `--compression`/`--recompress` flags, but I'm unsure how exactly this would look. Can anyone provide an example of what I should be doing here, or if it's even possible. 

Thanks!
-Dan

**Comment by ThomasWaldmann** [comment, marked as answer]
https://borgbackup.readthedocs.io/en/stable/usage/recreate.html

2nd example there in the code block maybe?

Nowadays I'ld advise you go with zstd for all compression levels except the fastest one, where lz4 might be even better.

**Reply by DizkoDan** [reply]
I wasn't sure if that assumed that it was previously compressed, based on the 1st command there. Does this need to be done per archive, or can it be done across the entire repo in one command? 

Thanks again,
-Dan

**Reply by ThomasWaldmann** [reply]
recreate works on a per-archive basis.

**Comment by DizkoDan** [comment]
That was easy, and it took a lot less time than I thought it would!

Before:
```
> ------------------------------------------------------------------------------
>                        Original size      Compressed size    Deduplicated size
> All archives:              799.18 GB            799.28 GB            184.53 GB
> 
>                        Unique chunks         Total chunks
> Chunk index:                  334519              2319330
```

After:
```
> ------------------------------------------------------------------------------
>                        Original size      Compressed size    Deduplicated size
> All archives:              797.35 GB            571.57 GB            131.22 GB
> 
>                        Unique chunks         Total chunks
> Chunk index:                  334399              2318517
```

Enabled zstd in my config and created a new archive:
```
> ------------------------------------------------------------------------------
>                        Original size      Compressed size    Deduplicated size
> This archive:              321.07 GB            222.95 GB             70.40 MB
> All archives:                1.12 TB            794.30 GB            131.29 GB
>                        Unique chunks         Total chunks
> Chunk index:                  337534              2822932
> ------------------------------------------------------------------------------
```

Thanks again!
